Suport serial type for sinks (#2343)
risingwavelabs/risingwave-docs: docs/guides/create-sink-kafka.md

)
```

> For data mapping, the serial type is supported, which is mapped into the 64-bit integer.

Reviewer: 64-bit integer -> int64

Reviewer 2: https://avro.apache.org/docs/1.11.1/specification/#primitive-types
> *long*: 64-bit signed integer

I think either long or 64-bit signed integer is better than int64.

> For data type mapping, the serial type is supported, which is mapped into the 64-bit signed integer.

Reviewer 3: ```suggestion
For data type mapping, the system supports mapping the serial type to a 64-bit signed integer.
```

)
```

> For data type mapping, the serial type is supported, which is mapped into the 64-bit signed integer.

Reviewer 3: ```suggestion
For data type mapping, the system supports mapping the serial type to a 64-bit signed integer.
```

### JSON specific parameters

> For data mapping, the serial type is supported. However, note that it is mapped into a JSON string like `0x05fb93d677c4e000` instead of a JSON number `431100738685689856`. This string form avoids JSON number precision issues with large int64 values, and you can still order by the fixed-length hexadecimal string to obtain the same order as the serial number (whereas variable-length string "12" sorts before "7").

Reviewer 3: ```suggestion
For data mapping, the system supports mapping the serial type to into a JSON string like `0x05fb93d677c4e000`,  not a JSON number like `431100738685689856`. This string form can avoid JSON number precision issues with large int64 values, and you can still order by the fixed-length hexadecimal string to obtain the same order as the serial number (whereas variable-length string "12" sorts before "7").
```

> For data mapping, the serial type is supported. However, note that it is mapped into a JSON string like `0x05fb93d677c4e000` instead of a JSON number `431100738685689856`. This string form avoids JSON number precision issues with large int64 values, and you can still order by the fixed-length hexadecimal string to obtain the same order as the serial number (whereas variable-length string "12" sorts before "7").

Reviewer 2: ```suggestion
For data mapping, the serial type is supported. However, note that it is mapped into a JSON string like `"0x05fb93d677c4e000"` instead of a JSON number `431100738685689856`. This string form avoids JSON number precision issues with large int64 values, and you can still order by the fixed-length hexadecimal string to obtain the same order as the serial number (whereas variable-length string "12" sorts before "7").
```

The quotes are intentional to indicate a string. Putting it into the context:
* `{"_row_id": 431100738685689856}` number form not adopted
* `{"_row_id": "0x05fb93d677c4e000"}` string form adopted
* `{"_row_id": 0x05fb93d677c4e000}` invalid JSON!

> For data mapping, the serial type is supported. But it is mapped into the string type. Because the serial type is often used as a key, precision issues with large int64 values in JSON can lead to unexpected problems. In our implementation, to preserve the original semantics of the serial type, we use hexadecimal strings with leading zeros. Adding leading zeros helps maintain numerical order, and using hexadecimal format prevents excessively long numbers from affecting display.

Reviewer: @xiangjinwu shall we give an example about hex encoded serial, people can be confused by leading zeros.

Reviewer 2: Yes I agree an example is more helpful than listing the considerations.

For data mapping, the serial type is supported. However, note that it is mapped into a JSON string like `"0x05fb93d677c4e000"` instead of a JSON number `431100738685689856`. This string form avoids JSON number precision issues with large int64 values, and you can still order by the fixed-length hexadecimal string to obtain the same order as the serial number (whereas variable-length string `"12"` sorts before `"7"`).
